pkg/promtail/targets/tailer.go

```diff
@@ -87,7 +87,14 @@ func (t *tailer) run() {
 	for {
 		select {
 		case <-positionWait.C:
-			err := t.markPosition()
+			fi, err := os.Stat(t.filename)
+			if err != nil {
+				level.Error(t.logger).Log("msg", "failed to stat log file being tailed, "+
+					"cannot report size or mark position", t.filename, "error", err)
+				continue
+			}
+			totalBytes.WithLabelValues(t.path).Set(float64(fi.Size()))
```

Comment:
By putting this here, we wont catch the case where `run` has exited or blocked.  Can we put this in `sync`?